ProcessMaker/Models/ProcessRequestToken.php

```diff
@@ -1152,4 +1171,64 @@ public function reassign($toUserId, User $requestingUser)
         $this->user->notify($notification);
         event(new ActivityAssigned($this));
     }
+
+    /**
+     * Determines the destination URL based on the element destination type specified in the definition.
+     *
+     * @return string|null
+     */
+    public function getElementDestinationAttribute(): ?string
+    {
+        $definition = $this->getDefinition();
+        $elementDestinationProp = $definition['elementDestination'] ?? null;
+        $elementDestination = null;
+        $elementDestinationType = null;
+
+        try {
+            $elementDestinationProp = json_decode($elementDestinationProp, true);
+            if (is_array($elementDestinationProp) && array_key_exists('type', $elementDestinationProp)) {
+                $elementDestinationType = $elementDestinationProp['type'];
+            }
+        } catch (Exception $e) {
+            return null;
+        }
+
+        switch ($elementDestinationType) {
```

Comment:
Here we can abstract this section into a method example getElementDestinationByType

Reply:
The method was improved